source/extensions/filters/http/adaptive_concurrency/adaptive_concurrency_filter.cc

```diff
@@ -45,7 +45,15 @@ Http::FilterHeadersStatus AdaptiveConcurrencyFilter::decodeHeaders(Http::HeaderM
   return Http::FilterHeadersStatus::Continue;
 }
 
-void AdaptiveConcurrencyFilter::encodeComplete() { deferred_sample_task_.reset(); }
+void AdaptiveConcurrencyFilter::encodeComplete() {
+  if (encoder_callbacks_->streamInfo().healthCheck()) {
```

Comment:
Wouldn't the health check filter always be installed in front of this filter? Would it be better to just check this in the decode path similar to the current runtime disable? That seems simpler to me.

Reply:
I wasn't sure if that would be the case. It would certainly make more sense to do it that way.

I'm working on the documentation PR now, so I'll make a note in there that this filter should come after the the healthcheck filter. I'll change this to be in the decode path.